File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Code/GestaoCarga/GestaoCargaBLL.cs
using JsonWebToken.Code.GestaoCarga;
using JsonWebToken.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace JsonWebToken.Code
{
  public class GestaoCargaBLL
  {
    public GestaoCargaModel selectGestaoCargaPorId(int id) {
      try
      {
        GestaoCargaDAL DAL = new GestaoCargaDAL();
        return DAL.selectGestaoCargaPorId(id);
      }
      catch (Exception e)
      {
        throw e;
      }
    }
    public List<GestaoCargaModel> selectGestaoCarga(DateTime dataRef, String tipo)
    {
      try
      {
        GestaoCargaDAL DAL = new GestaoCargaDAL();
        List<GestaoCargaModel> list = DAL.selectGestaoCarga(dataRef, tipo);
        return list;
      }
      catch (Exception e)
      {
        throw e;
      }
    }
  

    public void deleteGestaoCarga(int id)
    {
      try
      {
        GestaoCargaDAL DAL = new GestaoCargaDAL();
      DAL.deleteGestaoCarga(id);

      }
      catch (Exception e)
      {
        throw e;
      }
    }
    
    public void updateGestaoCarga(GestaoCargaModel gestaoCargaModel)
    {
      GestaoCargaDAL DAL = new GestaoCargaDAL();
      DAL.atualizarVersao(gestaoCargaModel);
    }

    public void insertGestaoCarga(DateTime dataRef )
    {
      try
      {
        GestaoCargaDAL DAL = new GestaoCargaDAL();
        DAL.insertGestaoCarga(dataRef);

      }
      catch (Exception e)
      {
        throw e;
      }
    }
  }

}

File: Backend/Administracao.ConciliacaoAPI/Code/GestaoCarga/GestaoCargaBLL.vb
Imports Martins.Conciliacao.Model

Namespace Code.GestaoCarga
  Public Class GestaoCargaBLL
    Public Function selectGestaoCargaPorId(id As Integer) As GestaoCargaModel
      Try
        Dim DAL As New GestaoCargaDAL()
        Return DAL.selectGestaoCargaPorId(id)
      Catch e As Exception
        Throw e
      End Try
    End Function
    Public Function selectGestaoCarga(dataRef As DateTime, tipo As [String]) As List(Of GestaoCargaModel)
      Try
        Dim DAL As New GestaoCargaDAL()
        Dim list As List(Of GestaoCargaModel) = DAL.selectGestaoCarga(dataRef, tipo)
        Return list
      Catch e As Exception
        Throw e
      End Try
    End Function


    Public Sub deleteGestaoCarga(id As Integer)
      Try
        Dim DAL As New GestaoCargaDAL()

        DAL.deleteGestaoCarga(id)
      Catch e As Exception
        Throw e
      End Try
    End Sub

    Public Sub updateGestaoCarga(gestaoCargaModel As GestaoCargaModel)
      Dim DAL As New GestaoCargaDAL()
      DAL.atualizarVersao(gestaoCargaModel)
    End Sub

    Public Sub insertGestaoCarga(dataRef As DateTime, tipo As String, codUnidade As Integer)
      Try
        Dim DAL As New GestaoCargaDAL()

        DAL.insertGestaoCarga(dataRef, tipo, codUnidade)
      Catch e As Exception
        Throw e
      End Try
    End Sub
  End Class

End Namespace

